### ISSUE TITLE
Create the "New visitors" key metric widget tile

### ISSUE THREAD

**Issue by jimmymadon** [google/site-kit-wp#6244]
## Feature Description

<img width="347" alt="Screenshot 2022-12-19 at 01 27 54" src="https://user-images.githubusercontent.com/1721319/210273030-09e39a10-480a-43d6-81b0-7311a353a072.png">

- Refer to the widget tables in the [design doc](https://docs.google.com/document/d/1k8NOo2xEHUQAcgYG0TB5sUjGcPiC8SWIalEdd9G8-9g/edit#bookmark=id.qz6oxctz8fe7) and the [PRD](https://docs.google.com/document/d/1JvJ2gKxaX2PcYOnmA3-xqz4hK_ZHxj8Rd0svq6S8tiU/edit#bookmark=id.w8o73hf5pz1f) explaining the sources of data and other details.
- [Figma mock for all widgets](https://www.figma.com/file/CdtrHB0HAjfGRWTnTMMQI0/Key-Metrics?node-id=3407%3A73377&t=mxlhDsRNGXcli8tk-0)

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* A new component should be created for the 'New visitors' Key metric widget and should be displayed within the Key Metrics widget area if it is selected (based on #6257 and #6313).
  * The component should reflect the screenshot supplied above, see the Figma link above for more detail.
* The following data should be fetched from the [Google Analytics Data API](https://developers.google.com/analytics/devguides/reporting/data/v1/api-schema) for the global date range selected on the Site Kit dashboard:
    * metric: `activeUsers`
    * dimension: `newVsReturning`
* Values should be displayed as:
    * `new` of  `(new + returning)` total visitors.
* The percentage change figure (with a red for negative or green for positive highlight) should be calculated based on the 'previous' period before the selected date range. 
* Existing GA4 datapoints and datastore infrastructure that is being added as part of the [GA4 Reporting Epic](https://github.com/google/site-kit-wp/issues/6253#workspaces/execution-5e185a810bd7fc358af69885/board?epics=Z2lkOi8vcmFwdG9yL0VwaWMvMTAxNTQyOQ) should be used where available.

## Implementation Brief
* Create `assets/js/modules/analytics-4/components/widgets/AnalyticsNewVisitorsWidget` which exports the `AnalyticsNewVisitorsWidget` functional component.
* It should render the info as per the designs in Figma.
    * Generic CSS class names (nothing "New visitors" specific class names should be used since the styles will be reused by other components).
        * For e.g the "googlesitekit-key-metrics__title" class name can be used for the title.
* Query the GA4 data store via the `getReport` selector with the metric and dimension as per the AC.
    * To get the selected date range, query the `core/user` data store via the `getDateRangeDates` with `compare` set to `true`.
* To get the % change, compare the values from `totals` for the current and previous date range. See below for the `Change` component.
* Create `assets/js/modules/analytics-4/components/common/Change.js` which exports the `Change` functional component.
    * It accepts the following props:
        * `previousValue`: `number`
        * `currentValue`: `number`
    * Based on the `previousValue` and `currentValue` props, it should calculate the percentage change using the `calculateChange` function (found in `assets/js/util/index.js`) and render the change using the `numFmt` function (found in `assets/js/util/i18n.js`).
    * It should render the change "badge", for e.g + 10.5% with the appropriate styles when the change is positive or negative. Refer to the [Figma designs](https://www.figma.com/file/CdtrHB0HAjfGRWTnTMMQI0/Key-Metrics?node-id=3407%3A73299&t=gAoifyRHk4k3xSO5-0) for the styles.
* Parse the data coming from `getReport` to display the values as per the AC.
* Look into the possibility to having a `assets/js/modules/analytics-4/components/common/MetricWidget.js` component which accepts `title` and `children` as props which can be re used by other components. The `MetricWidget ` component will be responsible to generate the widget markup and the title along with the rounded corners styles, and then `children` will be rendered below the `title`.
* Create `assets/js/modules/analytics-4/components/widgets/AnalyticsNewVisitorsWidget.stories.js` which contains the stories for the component.
* All styles should live in the `assets/sass/components/key-metrics` folder.

### Test Coverage
* Add tests for the `Change` and `AnalyticsNewVisitorsWidget` components.

## QA Brief

* With the `userInput` feature flag enabled, answer the questionnaire. For now, all widgets are displayed on the dashboard (which will change with future commits). As part of this issue, test the design and metric values of the `New Visitors` widget rendered in the Key Metrics section.
* Test the generic `Metric Tile Numeric` and the specific `New Visitors` stories within the Key Metrics widget section in Storybook.
* Note 1: The mobile version currently does not display Key Metric tiles in a single column. A new issue should be created for this.
* Note 2: There is no design for the loading state - we need to speak to UX and finalise a design and implement loading states for all widgets.
* Note 3: Designing the error state for widget tiles is out of the scope of this issue. A new issue should be created for this.

## Changelog entry

* Implement the `New visitors` key metric widget.

**Comment by tofumatt** [comment]
ACs here are good 👍🏻

**Comment by tofumatt** [comment]
IB ✅ 

I like the child component idea. We might want to call it something less generic, like `MetricWidget` or something, but looks good 👍🏻

**Comment by asvinb** [comment]
Thanks @tofumatt , I've tweaked the IB by replacing `Widget` by `MetricWidget` 👍

**Comment by hussain-t** [comment]
Key Metric widget components (skeleton) are created in https://github.com/google/site-kit-wp/issues/6313. So we don't have to create the `AnalyticsNewVisitorsWidget` component.

**Comment by hussain-t** [comment]
@asvinb @tofumatt, according to the AC, the criteria to render the widget tile is missing in the IB:
>A new component should be created for the 'New visitors' Key metric widget and should be displayed within the Key Metrics widget area if it is selected (based on https://github.com/google/site-kit-wp/issues/6257

It should check if the selected metric depends on GA4 using the `getKeyMetrics` selector and render the elements.

**Comment by marrrmarrr** [comment]
Hi @jimmymadon, I was more thinking of option (2), as I don't know if most users would understand the comparison of percentages very easily. We're mostly aiming to give them an idea of whether the portion of new visitors is increasing or decreasing, whereas option (1) would be more about whether the rate of growth of new visitors.

**Comment by wpdarren** [comment]
@jimmymadon while this is in Execution, I am seeing a difference between the Site Kit tile for New visitors to what I see on Analytics.   I added a comment on [this ticket](https://github.com/google/site-kit-wp/issues/6243#issuecomment-1589050392) because this has an impact on the calculation of the 'Loyal visitors' percentage.  

Would you be able to take a look at it?

**Comment by techanvil** [comment]
@mohitwp @jimmymadon as discussed on Slack I've created a followup issue to address Question 2: https://github.com/google/site-kit-wp/issues/7172, which can therefore be identified as a known issue here.

**Comment by mohitwp** [comment]
### QA Update ⚠️ 

@jimmymadon @eugene-manuilov 

1 > I'm not able to found any data as you mentioned in the comment. 

`You can just compare the KMW data in Site Kit using GA4 Admin console for the exact same property and date range. In Analytics console, in the side panel, see Reports -> Audience -> Users -> New vs Returning.`

2> In QAB it is mentioned that dimension: `newVsReturning`. But I'm not getting this dimension under dimension list. 

![image](https://github.com/google/site-kit-wp/assets/94359491/000bdbb7-4c90-4ac0-8cbf-b85c5a8bc9fd)

3> I found one matching value under Monetization>Retention.  If we consider it values then as per formula  `new of (new + returning) total visitors`` current value we are showing in widget is not correct.

![image](https://github.com/google/site-kit-wp/assets/94359491/0d7a959a-f496-4bba-9a60-e085d1410254)

![image](https://github.com/google/site-kit-wp/assets/94359491/dc61864f-abec-496c-9609-e15fbd71e1a0)

4> Also, can we display abbreviation of numbers in small letters. k instead of K. 

![image](https://github.com/google/site-kit-wp/assets/94359491/51870cdd-c268-411d-a7d3-6d35492cdeeb)
